sandbox/lua/encoders/opentsdb_http.lua

```diff
@@ -0,0 +1,190 @@
+last Timestamp and the current one (flush_delta) (encoders currently have no
+support for timer_event()s)
+
+Config:
+
+- flush_count (number, default 1)
+    Flush the buffer every 'flush_count' messages.
+
+- flush_delta (number, default 0)
+    Flush the buffer if 'flush_delta' seconds have elapsed since the last
+    Timestamp and the current one.
+
+- tag_prefix (string, optional, default "")
+    Convert Fields to OpenTSDB tags if they match the tag_prefix.
+    (tag_prefix is stripped from the tag name)
+
+- tag_list (string, optional, default "")
+    A list of fields (separated by spaces) which will be converted into tags.
+    You can use either "tag_list", "tag_prefix" or both.
+
+- skip_fields (string, optional, default "")
+    A list of fields (separated by spaces) which will be dropped from processing.
+
+- type_as_prefix (boolean, optional, default true)
+    Prepend metrics when encoding them to OpenTSDB format by Type field.
+
+- ts_from_message (boolean, optional, default false)
+    Use the Timestamp field (otherwise "now")
+
+- add_hostname_if_missing (boolean, optional, default false)
+    If no 'host' tag has been seen, append one with the value of the
+    Hostname field.  Deprecated in favour of using the 'fieldfix' filter.
+
+
+*Example Heka Configuration*
+.. code-block:: ini
+
+    [OpentsdbEncoder]
+    type = "SandboxEncoder"
+    filename = "lua_encoders/opentsdb_http.lua"
+
+      [OpentsdbEncoder.config]
+      ts_from_message = true
+      add_hostname_if_missing = true
+      type_as_prefix = true
+      tag_prefix = "tag_"
+      tag_list = "env dc"
+      skip_fields = "Pid"
+
+    [opentsdb]
+    type = "HttpOutput"
+    message_matcher = "Fields[MetricOpentsdb] != NIL"
+    address = "http://opentsdb.example.com:4242/api/put"
+    encoder = "OpentsdbEncoder"
+
+*Example Output*
+.. code-block:: json
+    [{"value":0.05,"timestamp":1443535912,"metric":"stats.loadavg.15MinAvg","tags":{"env":"test","host":"example.com"}},{"value":2,"timestamp":1443535912,"metric":"stats.loadavg.NumProcesses","tags":{"env":"test","host":"example.com"}},{"value":0.01,"timestamp":1443535912,"metric":"stats.loadavg.5MinAvg","tags":{"env":"test","host":"example.com"}},{"value":0,"timestamp":1443535912,"metric":"stats.loadavg.1MinAvg","tags":{"env":"test","host":"example.com"}}]
+
+--]=]
+
+require "cjson"
+require "table"
+require "string"
+require "os"
+
+local l = require 'lpeg'
+l.locale(l)
+
+local flush_count     = read_config("flush_count") or 1
+local flush_delta     = read_config("flush_delta") or 0
+local tag_prefix      = read_config("tag_prefix")
+local tag_list_str    = read_config("tag_list") or ""
+local skip_fields_str = read_config("skip_fields") or ""
+local type_as_prefix  = read_config("type_as_prefix")
+local ts_from_message = read_config("ts_from_message")
+local add_hostname    = read_config("add_hostname_if_missing")
+_PRESERVATION_VERSION = read_config("preservation_version") or 1
```

Comment:
Our convention is to put the _PRESERVATION_VERSION stuff as the first lines of the module, immediately after the documentation comment. Also, you'll want to add in a value that you can increment in the source code, for times when changing the source code of the encoder invalidate the saved data. You can see an example of how this should be done in the [HTTP Status filter](https://github.com/mozilla-services/heka/blob/v0.10.0b1/sandbox/lua/filters/http_status.lua#L49).